doc/source/developer/specs/data-source.md

```diff
@@ -24,22 +24,39 @@
 
 - `source`: Any, required
     + The location of the source. Depending on the child node this can be a filepath, numpy array, or dictionary as a few examples.
-- `interpolation`: `Interpolator()`  class
+- `interpolation`: 
+    - `Interpolator()` - NearestNeighbor
+    - Dict({`dim` (not stacked): `Interpolator()` NearestNeighbor})
+    - string - Enum(`interpolate.INTERPOLATION_OPTIONS`)
+    - Dict({`dim` (not stacked): string - Enum(`interpolate.INTERPOLATION_OPTIONS`)})
 - `coordinate_index_type`: Enum('list','numpy','xarray','pandas'). By default this is `numpy`
 - `nan_vals`: List
     + list of values from source data that should be interpreted as 'no data' or 'nans' (replaces `no_data_vals`)
 
+#### Private Members
+
+- `_interpolator` - interpolator chosen from `interpolator` and `interpolation`
+
+*TODO* : the names of these memebers will be changed
+
+- `_requested_coordinates` = tl.Instance(Coordinates, allow_none=True)
+- `_requested_source_coordinates` = tl.Instance(Coordinates)
+- `_requested_source_coordinates_index` = tl.List()
+- `_requested_source_data` = tl.Instance(UnitsDataArray)
+
 #### Properties
 
 #### Contructor
 
 - FUTURE: After implementing a limiter on the request size, implement:
     + Take one input (i.e. `evaluate`) that will automatically execute the datasource at the native_coordinates on contruction. This will allow a shortcut when you just want to load a simple data source for processing with other more complication data sources
+- Choose a default interpolation option if neither interpolator or interpolation is defined
```

Comment:
is `interpolator == _interpolator`?

Reply:
this should read 'Choose a default interpolation option if `interpolation` is defined'

I have not pushed changes to DataSource spec - pushing now